llm_studio/src/datasets/text_rlhf_modeling_ds.py

```diff
@@ -0,0 +1,67 @@
+import logging
+from typing import Any, Dict
+
+import numpy as np
+import pandas as pd
+import torch
+
+from llm_studio.src.datasets.text_causal_language_modeling_ds import (
+    CustomDataset as CausalLMCustomDataset,
+)
+from llm_studio.src.datasets.text_utils import get_texts
+
+logger = logging.getLogger(__name__)
+
+
+class CustomDataset(CausalLMCustomDataset):
+    def __init__(self, df: pd.DataFrame, cfg: Any, mode: str = "train"):
+        assert (
+            cfg.dataset.system_column == "None"
+        ), "RLHF is not compatible with system column."
+        super().__init__(df, cfg, mode)
+        self.raw_prompts = get_texts(df, self.cfg, separator="")
+
+    def __getitem__(self, idx: int) -> Dict:
+        """Reads a single text observation."""
+        sample = super().__getitem__(idx)
+        idx = self.indices[idx]
+        sample["reward_model_prompt_text"] = self.get_reward_model_prompt_text(idx)
+        return sample
+
+    def get_labels(self, encodings):
+        if self.mode != "train":  # no labels required for RLHF during training
```

Comment:
shouldn't that be `==` then?

Reply:
Good catch. One could also always return labels, as we do not (yet?) support RLHF on prompt dataset, only.